src/adapters/bam/llm.ts

```diff
@@ -221,6 +225,28 @@ export class BAMLLM extends LLM<BAMLLMOutput, BAMLLMGenerateOptions> {
     }
   }
 
+  async embed(input: LLMInput[], options?: EmbeddingOptions): Promise<EmbeddingOutput> {
+    const results = await Promise.all(
+      chunk(input, MAX_EMBEDDING_INPUTS).map(async (texts) => {
```

Comment:
Is chunking needed?

Reply:
I think the chunking is related to the concurrency limit on the embeddings endpoint which is 20: https://bam.res.ibm.com/docs/api-reference#limits

Given the state of BAM, I dont think we should spend too much time on this one.

Reply:
i agree.

Reply:
I see, the SDK does not handle chunking.